src/frequenz/sdk/timeseries/formula_engine/_formula_generators/_battery_power_formula.py

```diff
@@ -62,15 +65,38 @@ def generate(
         component_graph = connection_manager.get().component_graph
 
         battery_inverters = list(
-            next(iter(component_graph.predecessors(bat_id))) for bat_id in component_ids
+            frozenset(
+                filter(
+                    component_graph.is_battery_inverter,
+                    component_graph.predecessors(bat_id),
+                )
+            )
+            for bat_id in component_ids
```

Comment:
in a setup with 1 inverter and 2 batteries, there will be 2 frozensets with the same inverter.  that would lead to 2x power reporting, no?

Reply:
Good catch. Added a test to catch this and fixed it, too